Update for ITK 5.1rc2

hjmjohnson opened the pull request:
Object MD5=b4ed0e70fb4f381c7a3967c650b0a824 not found at:

    https://insightsoftwareconsortium.github.io/ITKTestingData/MD5/b4ed0e70fb4f381c7a3967c650b0a824 ("HTTP response code said error")
    https://data.kitware.com:443/api/v1/file/hashsum/MD5/b4ed0e70fb4f381c7a3967c650b0a824/download ("HTTP response code said error")
    https://itk.org/files/ExternalData/MD5/b4ed0e70fb4f381c7a3967c650b0a824 ("Timeout was reached")
    https://slicer.kitware.com/midas3/api/rest?method=midas.bitstream.download&checksum=b4ed0e70fb4f381c7a3967c650b0a824&algorithm=MD5 ("Timeout was reached")

hjmjohnson committed:
STYLE: Replace integer literals which are cast to bool.

Finds and replaces integer literals which are cast to bool.

SRCDIR= #My local SRC
BLDDIR= #My local BLD

cd
run-clang-tidy.py -extra-arg=-D__clang__ -checks=-*,modernize-use-bool-literals  -header-filter=.* -fix

hjmjohnson committed:
STYLE: Prefer = default to explicitly trivial implementations

This check replaces default bodies of special member functions with
= default;. The explicitly defaulted function declarations enable more
opportunities in optimization, because the compiler might treat
explicitly defaulted functions as trivial.

Additionally, the C++11 use of = default more clearly expreses the
intent for the special member functions.

hjmjohnson committed:
STYLE: Remove redundant void argument lists

Find and remove redundant void argument lists.

hjmjohnson committed:
DOC: Update copyright assignment to NumFOCUS

The mission of NumFOCUS is to promote open
practices in research, data, and scientific
computing.

https://numfocus.org

dzenanz reviewed (commented):
Commit title misleading, otherwise looks good.

hjmjohnson commented on test/CMakeLists.txt:
```diff
@@ -17,6 +17,7 @@ itk_add_test(NAME TwoProjection2D3DRegistrationDownSizedCTTest
     DATA{Input/BoxheadCT.img,BoxheadCT.hdr}
   )
 
+if(0) #HACK File missing from download
```
@dzenanz  I don't think we should use this PR, until the tests are fixed.  Currently the data is not available for download.  How can we restore these data sets?

hjmjohnson committed:
ENH: Add .gitattributes to allow running ITK clang-formatting scripts

```
git filter-branch -f \
--tree-filter "~/ITK/Utilities/Maintenance/clang-format.bash --clang-format ~/Dashboard/src/ITK-clang11/clang-format-Linux --tracked" \
    master..
```

hjmjohnson committed:
ENH: Set ITKGitTag to v5.1rc02 for azure CI

A newer version of ITK is needed for CI
build to succeed.

dzenanz commented:
I guess I will have to look into this either tomorrow or next week, as workday is almost over.

hjmjohnson committed:
ENH: Fix circleci pathing failure.

hjmjohnson committed:
ENH: Prefer https vs http for ctest submission.

dzenanz commented on test/azure-pipelines.yml:
```diff
@@ -75,11 +77,16 @@ jobs:
       curl -L https://raw.githubusercontent.com/InsightSoftwareConsortium/ITK/dashboard/itk_common.cmake -O
     displayName: 'Fetch CTest driver script'
 
+  - bash: |
```
What is this for? In order to turn module template into a module, cookiecutter needs to be applied once. I don't think we need to apply it again here. Same for instances below.

hjmjohnson replied:
@dzenanz I was following the example given in `ITKModuleTemplate`

These lines come from synchronizing with:
https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/master/azure-pipelines.yml#L80

dzenanz replied:
If you try to synchronize with it directly, some weirdness sneaks in, e.g. `ITKModuleTemplate` is used as a name instead of `ITKTwoProjectionRegistration`.

You should invoke cookie-cutter and enter this module's information (name `ITKTwoProjectionRegistration` etc). The result should then be used for syncing.

hjmjohnson committed:
ENH: Allow building shared library.

hjmjohnson committed:
COMP: Synchronize with template builder scripts.

hjmjohnson committed:
BUG: Incorrect test library variable used.

Test driver did not compile due to empty list
of dependant link libraries.
